src/Searchlight.MongoDB/MongoDbExecutor.cs

```diff
@@ -70,10 +72,22 @@ public static FilterDefinition<T> BuildMongoFilter<T>(List<BaseClause> clauses)
                             case OperationType.LessThanOrEqual:
                                 return Builders<T>.Filter.Lte(criteria.Column.FieldName, criteria.Value);
                             case OperationType.Contains:
-                                return Builders<T>.Filter.Text(criteria.Column.FieldName, criteria.Value.ToString());
+                                return Builders<T>.Filter.Regex(criteria.Column.FieldName,
+                                    new BsonRegularExpression("/" + criteria.Value + "/"));
+                            case OperationType.StartsWith:
+                                return Builders<T>.Filter.Regex(criteria.Column.FieldName,
+                                    new BsonRegularExpression("/^" + criteria.Value + "/"));
+                            case OperationType.EndsWith:
+                                return Builders<T>.Filter.Regex(criteria.Column.FieldName,
+                                    new BsonRegularExpression("/" + criteria.Value + "$/"));
                             default:
                                 throw new NotImplementedException();
```

Comment:
Are there any other operation types?

Reply:
I don't think so, there's an unknown one but the default case should catch that